### ISSUE TITLE
Incorrect indent when else statement has no curly braces

### ISSUE THREAD

**Issue by RubenVerborgh** [eslint/eslint#2343]
**The `indent` error is incorrectly raised when an `if` statement uses curly braces, but its `else` counterpart does not.**

The following example has proper indentation:
```JavaScript
/*eslint "indent": 2*/
/*eslint "curly": 0*/
(function () {
    "use strict";
    switch (this.a) {
        case "a":
            if (this.a === this.b) {
                this.a = "a";
                this.b = "b";
            }
            else
                this.a = "c";  // this line triggers an error
            break;
    }
})();
```
but results in
```
12:8  error  Expected indentation of 8 characters  indent
```

The following example has improper indentation:
```JavaScript
/*eslint "indent": 2*/
/*eslint "curly": 0*/
(function () {
    "use strict";
    switch (this.a) {
        case "a":
            if (this.a === this.b) {
                this.a = "a";
                this.b = "b";
            }
            else
        this.a = "c";  // this line should trigger an error
            break;
    }
})();
```
but does not trigger any error.

Even when curly braces are used, the error is off. This examples has proper indentation:

```JavaScript
/*eslint "indent": 2*/
(function () {
    "use strict";
    switch (this.a) {
        case "a":
            if (this.a === this.b) {
                this.a = "a";
                this.b = "b";
            }
            else {
                this.a = "c";
            }
            break;
    }
})();
```
triggers:
```
10:8   error  Expected indentation of 8 characters   indent
11:12  error  Expected indentation of 12 characters  indent
12:8   error  Expected indentation of 8 characters   indent
```

**Comment by RubenVerborgh** [comment]
Likely related to #2248.

**Comment by lo1tuma** [comment]
Maybe the same bug as described in #1799.
